Marlin/Marlin_main.cpp

```diff
@@ -5832,10 +5832,17 @@ inline void gcode_M17() {
   ) {
     if (move_away_flag) return false; // already paused
 
-    if (!DEBUGGING(DRYRUN) && thermalManager.tooColdToExtrude(active_extruder) && unload_length > 0) {
-      SERIAL_ERROR_START;
-      SERIAL_ERRORLNPGM(MSG_TOO_COLD_FOR_M600);
-      return false;
+    if (!DEBUGGING(DRYRUN) && unload_length > 0) {
+      #if ENABLED(PREVENT_COLD_EXTRUSION)
+        if (!thermalManager.allow_cold_extrude && 
+            thermalManager.degTargetHotend(active_extruder) < thermalManager.extrude_min_temp) {
+          SERIAL_ERROR_START;
+          SERIAL_ERRORLNPGM(MSG_TOO_COLD_FOR_M600);
+          return false;
+        }
+      #endif
```

Comment:
This was the very beginning of gcode_M600 prior to #6407 being merged:

```
    if (!DEBUGGING(DRYRUN) && thermalManager.tooColdToExtrude(active_extruder)) {
      SERIAL_ERROR_START;
      SERIAL_ERRORLNPGM(MSG_TOO_COLD_FOR_M600);
      return;
    }
```

It didn't check to see if the target temperature was reasonable and if the nozzle wasn't warm enough, it would just bail. That's exactly how advance pause worked prior to the addition of `ensure_safe_temperature`. This change adds some additional checks to ensure that the target temperature is above the limit (if there is a limit) and then waits for the nozzle to heat up prior to continuing, which is what you requested in your latest comments in #6407.